db/push.go

```diff
@@ -0,0 +1,96 @@
+package db
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	nomsjson "roci.dev/diff-server/util/noms/json"
+)
+
+// These request/response types might want to live in a different location, eg 
+// in diff-server/serve/types.
+type Mutation struct {
+	ID   uint64          `json:"id"`
+	Name string          `json:"name"`
+	Args json.RawMessage `json:"args"`
+}
+
+type BatchPushRequest struct {
+	ClientID  string     `json:"clientId"`
+	Mutations []Mutation `json:"mutations"`
+}
+
+type MutationInfo struct {
+	ID    string `json:"id"`
+	Error string `json:"error"`
+}
+
+type BatchPushResponse struct {
+	MutationInfos []MutationInfo `json:"mutationInfos"`
+}
+
+type BatchPushInfo struct {
+	HTTPStatusCode    int               `json:"httpStatusCode"`
```

Comment:
Does caller require access to `HTTPStatusCode` and `error`? Seems like it would only need either the `BatchPushResponse` if there was one, or `error` (something else bad happened). In which case, we could get rid of this struct.

Reply:
hmmm. i think both are useful? you want status code for programmatic use and the response body i think can be useful to humans.